code/modules/mob/living/carbon/human/appearance.dm

```diff
@@ -231,8 +312,52 @@
 						valid_facial_hairstyles += facialhairstyle
 					continue
 		else //If the user is not a species who can have robotic heads, use the default handling.
-			if(!(species.name in S.species_allowed))
+			if(!(H.species.name in S.species_allowed)) //If the user's head is not of a species the facial hair style allows, skip it. Otherwise, add it to the list.
 				continue
 			valid_facial_hairstyles += facialhairstyle
 
 	return valid_facial_hairstyles
+
+/mob/living/carbon/human/proc/generate_valid_head_accessories()
+	var/list/valid_head_accessories = new()
+	var/obj/item/organ/external/head/H = get_organ("head")
+	for(var/head_accessory in head_accessory_styles_list)
+		var/datum/sprite_accessory/S = head_accessory_styles_list[head_accessory]
+
+		if(!(H.species.name in S.species_allowed)) //If the user's head is not of a species the head accessory style allows, skip it. Otherwise, add it to the list.
+			continue
+		valid_head_accessories += head_accessory
+
+	return valid_head_accessories
+
+/mob/living/carbon/human/proc/generate_valid_markings()
+	var/list/valid_markings = new()
+	var/obj/item/organ/external/head/H = get_organ("head")
+	for(var/marking in marking_styles_list)
+		var/datum/sprite_accessory/S = marking_styles_list[marking]
+
+		if(!(species.name in S.species_allowed)) //If the user's head is not of a species the marking style allows, skip it. Otherwise, add it to the list.
+			continue
+		if(H.species.flags & ALL_RPARTS) //If the user is a species that can have a robotic head...
```

Comment:
Dives into a check specifically for players of species that have cybernetic.whatever heads that only lists markings compatible with their model of head.